### PR TITLE
[JENKINS-58684] use sse gateway 1.20

### PR DESCRIPTION
**Pull request by olamy**
Signed-off-by: olivier lamy <olamy@apache.org>

# Description

See [JENKINS-58684](https://issues.jenkins-ci.org/browse/JENKINS-58684).

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [ ] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -147,13 +147,16 @@
   <repositories>
     <repository>
       <id>repo.jenkins-ci.org</id>
-      <url>http://repo.jenkins-ci.org/public/</url>
+      <url>https://repo.jenkins-ci.org/public/</url>
+      <snapshots>
+        <enabled>true</enabled>
```
[the remark this answers is not in the document]
Comment (olamy) [reply pom.xml]: yes correct. I was using that because I tested with a SNAPSHOT version of sse.
If you use only fixed version this doesn't change anything

**Comment by halkeye** [comment]
[2019-08-29T01:49:33.903Z] [WARNING] Rule 5: org.apache.maven.plugins.enforcer.RequireUpperBoundDeps failed with message:

[2019-08-29T01:49:33.903Z] Failed while enforcing RequireUpperBoundDeps. The error(s) are [

[2019-08-29T01:49:33.903Z] Require upper bound dependencies error for commons-io:commons-io:2.4 paths to dependency are:

[2019-08-29T01:49:33.903Z] +-io.jenkins.blueocean:blueocean-events:1.18.0-rc4045.ccfa32645162

[2019-08-29T01:49:33.903Z]   +-org.jenkins-ci.plugins:sse-gateway:1.20

[2019-08-29T01:49:33.903Z]     +-commons-io:commons-io:2.4

[2019-08-29T01:49:33.903Z] and

[2019-08-29T01:49:33.903Z] +-io.jenkins.blueocean:blueocean-events:1.18.0-rc4045.ccfa32645162

[2019-08-29T01:49:33.903Z]   +-org.jenkins-ci.main:jenkins-core:2.164.1

[2019-08-29T01:49:33.903Z]     +-commons-io:commons-io:2.6

[2019-08-29T01:49:33.903Z] and

[2019-08-29T01:49:33.903Z] +-io.jenkins.blueocean:blueocean-events:1.18.0-rc4045.ccfa32645162

[2019-08-29T01:49:33.903Z]   +-org.jenkins-ci.main:jenkins-core:2.164.1

[2019-08-29T01:49:33.903Z]     +-org.jenkins-ci.main:cli:2.164.1

[2019-08-29T01:49:33.903Z]       +-commons-io:commons-io:2.6

[2019-08-29T01:49:33.903Z] and

[2019-08-29T01:49:33.903Z] +-io.jenkins.blueocean:blueocean-events:1.18.0-rc4045.ccfa32645162

[2019-08-29T01:49:33.903Z]   +-org.jenkins-ci.main:jenkins-core:2.164.1

[2019-08-29T01:49:33.903Z]     +-org.jenkins-ci:crypto-util:1.1

[2019-08-29T01:49:33.903Z]       +-commons-io:commons-io:1.4

[2019-08-29T01:49:33.903Z] and

[2019-08-29T01:49:33.903Z] +-io.jenkins.blueocean:blueocean-events:1.18.0-rc4045.ccfa32645162

[2019-08-29T01:49:33.903Z]   +-org.jenkins-ci.main:jenkins-core:2.164.1

[2019-08-29T01:49:33.903Z]     +-org.jenkins-ci:bytecode-compatibility-transformer:2.0-beta-2

[2019-08-29T01:49:33.903Z]       +-commons-io:commons-io:2.4

[2019-08-29T01:49:33.903Z] and

[2019-08-29T01:49:33.903Z] +-io.jenkins.blueocean:blueocean-events:1.18.0-rc4045.ccfa32645162

[2019-08-29T01:49:33.903Z]   +-org.jenkins-ci.main:jenkins-core:2.164.1

[2019-08-29T01:49:33.903Z]     +-commons-fileupload:commons-fileupload:1.3.1-jenkins-2

[2019-08-29T01:49:33.903Z]       +-commons-io:commons-io:2.2

[2019-08-29T01:49:33.903Z] and

[2019-08-29T01:49:33.903Z] +-io.jenkins.blueocean:blueocean-events:1.18.0-rc4045.ccfa32645162

[2019-08-29T01:49:33.903Z]   +-org.jenkins-ci.main:jenkins-core:2.164.1

[2019-08-29T01:49:33.903Z]     +-org.kohsuke.stapler:stapler-jrebel:1.256

[2019-08-29T01:49:33.903Z]       +-org.kohsuke.stapler:stapler:1.256

[2019-08-29T01:49:33.903Z]         +-commons-io:commons-io:2.4

[2019-08-29T01:49:33.903Z] ]

**Comment by olamy** [comment]
@halkeye ?? where do you get that? looks green here https://ci.blueocean.io/blue/organizations/jenkins/blueocean/detail/JENKINS-58684-sse-gateway-1.20/2/

**Comment by halkeye** [comment]
https://ci.jenkins.io/blue/organizations/jenkins/Plugins%2Fblueocean-plugin/detail/JENKINS-58684-sse-gateway-1.20/2/pipeline

**Comment by olamy** [comment]
LOL and in both case it's ` Apache Maven 3.5.4 `

**Comment by halkeye** [comment]
I was able to reproduce it with the same command
```
mvn --batch-mode --show-version --errors -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn  --update-snapshots -Dmaven.test.failure.ignore -Dset.changelist -Djenkins.version=2.164.1 -Daccess-modifier-checker.failOnError=false -Djava.level=8 -DskipTests clean install -Dignore.dirt
```

```
 (ssh) cb-gavinmogan   JENKINS-58684-sse-gateway-1.20 ? ⍟4  …/git/jenkinsci/blueocean-plugin  mvn -version
Apache Maven 3.6.1 (d66c9c0b3152b2e69ee9bac180bb8fcc8e6af555; 2019-04-04T12:00:29-07:00)
Maven home: /home/gmogan/.sdkman/candidates/maven/current
Java version: 1.8.0_202, vendor: Azul Systems, Inc., runtime: /home/gmogan/.sdkman/candidates/java/8.0.202-zulu/jre
Default locale: en_CA, platform encoding: UTF-8
OS name: "linux", version: "4.15.0-58-generic", arch: "amd64", family: "unix"

```

**Comment by olamy** [comment]
well ok I can fix that.
But that's crap to have such failure because of a different command line usage. 
how a normal contributor could know that...
this should fail because of a normal build (i.e no extra parameters..)

**Comment by halkeye** [comment]
Yea you'll have to ask upstream why that happened. Maybe the new parent pom or something. It used to report an error.

I'm thinking it might actually be a ci issue, I think they changed the enforcer settings to ignore test deps?

**Comment by olamy** [comment]
@halkeye maybe we can merge it now?

**Comment by halkeye** [comment]
```java.lang.AssertionError: nodes != 9:[PipelineNodeImpl{node=io.jenkins.blueocean.rest.impl.pipeline.FlowNodeWrapper[id=6,displayName=first-solo,type=STAGE], edges=[io.jenkins.blueocean.rest.impl.pipeline.PipelineNodeImpl$EdgeImpl@166b42ea], status=io.jenkins.blueocean.rest.impl.pipeline.NodeRunStatus@5d6e6c83}, PipelineNodeImpl{node=io.jenkins.blueocean.rest.impl.pipeline.FlowNodeWrapper[id=12,displayName=parent,type=STAGE], edges=[io.jenkins.blueocean.rest.impl.pipeline.PipelineNodeImpl$EdgeImpl@6adc1172], status=io.jenkins.blueocean.rest.impl.pipeline.NodeRunStatus@6fe07eb2}, PipelineNodeImpl{node=io.jenkins.blueocean.rest.impl.pipeline.FlowNodeWrapper[id=20,displayName=single-stage,type=STAGE], edges=[], status=io.jenkins.blueocean.rest.impl.pipeline.NodeRunStatus@23d01db6}]```

Yea, the failed test is flakey, has been for a few months, maybe #2017 will fix it

But yea, I approved it, go ahead and merge

**Pull request merged**